common-app-lib/src/cmr/common_app/services/kms_lookup.clj

```diff
@@ -280,95 +296,122 @@
   and returns the KMS keyword map as its stored in the cache. Returns nil if a keyword is not found.
   Comparison is made case insensitively."
   [context keyword-scheme umm-c-keyword]
-  (let [umm-c-keyword (csk-extras/transform-keys csk/->kebab-case umm-c-keyword)
-        format-map (if (string? umm-c-keyword)
-                     {:short-name umm-c-keyword}
-                     {:short-name (:format umm-c-keyword)})
-        comparison-map (normalize-for-lookup format-map (kms-scheme->fields-for-umm-c-lookup
-                                                         keyword-scheme))
-        umm-c-cache (hash-cache/context->cache context kms-umm-c-cache-key)
-        [tm value] (util/time-execution (hash-cache/get-value umm-c-cache kms-umm-c-cache-key keyword-scheme))
-        _ (rl-util/log-redis-read-complete "lookup-by-umm-c-keyword-data-format" kms-umm-c-cache-key tm)
-        value (if value
-                value
-                (let [_ (load-cache-if-necessary context umm-c-cache kms-umm-c-cache-key)
-                      [tm vlue] (util/time-execution (hash-cache/get-value umm-c-cache kms-umm-c-cache-key keyword-scheme))]
-                  (rl-util/log-redis-read-complete "lookup-by-umm-c-keyword-data-format" kms-umm-c-cache-key tm)
-                  vlue))]
-    (get-in value [comparison-map])))
+  (try
+    (let [umm-c-keyword (csk-extras/transform-keys csk/->kebab-case umm-c-keyword)
+          format-map (if (string? umm-c-keyword)
+                       {:short-name umm-c-keyword}
+                       {:short-name (:format umm-c-keyword)})
+          comparison-map (normalize-for-lookup format-map (kms-scheme->fields-for-umm-c-lookup
+                                                            keyword-scheme))
+          umm-c-cache (hash-cache/context->cache context kms-umm-c-cache-key)
+          [tm value] (util/time-execution (hash-cache/get-value umm-c-cache kms-umm-c-cache-key keyword-scheme))
+          _ (rl-util/log-redis-read-complete "lookup-by-umm-c-keyword-data-format" kms-umm-c-cache-key tm)
+          value (if value
+                  value
+                  (let [_ (load-cache-if-necessary context umm-c-cache kms-umm-c-cache-key)
+                        [tm vlue] (util/time-execution (hash-cache/get-value umm-c-cache kms-umm-c-cache-key keyword-scheme))]
+                    (rl-util/log-redis-read-complete "lookup-by-umm-c-keyword-data-format" kms-umm-c-cache-key tm)
+                    vlue))]
+      (get-in value [comparison-map]))
+    (catch Exception e
+      (if (clojure.string/includes? (ex-message e) "Carmine connection error")
+        (do
+          (error "lookup-by-umm-c-keyword-data-format found redis carmine exception. Will return nil result." e)
+          nil)
```

Comment:
Remove the (do and the nil)
(error returns nil.

Please do this in the kms_lookup namespace as well.